### PR TITLE
multiple owner support in muc_light.

### PR DESCRIPTION
**Pull request by jaspreet-android**
This PR addresses #

Proposed changes include:
* I want to support multiple owners in Mongooseim so that more than 1 owner can control the room or the owner can give equal rites to another member.
* I have added a test that the existing owner can add another owner or owners.

### CODE REVIEW

**Comment by chrzaszcz** [comment]
Hi @jaspreet-android, thank you for your contribution.

We can soon take a look into this, but just from a quick glance I can see that:
- Introducing a new module only to handle a single option in MUC Light is not a good idea. The configuration of `mod_muc_light` itself should be extended instead.
- There should be Markdown documentation in `docs` for the new option, and tests in `test/config_parser_SUITE.erl` for it.

We could follow with a more detailed code review, but this might take some time, since our team has a lot of high-priority tasks at the moment. The next step could be either a fix from you, or we could refine the PR ourselves, but this might take some time.

**Comment by jaspreet-android** [comment]
> Hi @jaspreet-android, thank you for your contribution.
> 
> We can soon take a look into this, but just from a quick glance I can see that:
> 
> * Introducing a new module only to handle a single option in MUC Light is not a good idea. The configuration of `mod_muc_light` itself should be extended instead.
> * There should be Markdown documentation in `docs` for the new option, and tests in `test/config_parser_SUITE.erl` for it.
> 
> We could follow with a more detailed code review, but this might take some time, since our team has a lot of high-priority tasks at the moment. The next step could be either a fix from you, or we could refine the PR ourselves, but this might take some time.

I will try to make it will muclight config only, I will add doc for the same.

File: big_tests/tests/muc_light_SUITE.erl [hunk 1]
```diff
@@ -792,6 +796,21 @@ remove_and_add_users(Config) ->
...
+    escalus:send(Alice, stanza_aff_set(?ROOM, AffUsersChanges1)),
+    verify_aff_bcast([{Alice, owner}], AffUsersChanges1),
+    escalus:assert(is_iq_result, escalus:wait_for_stanza(Alice)),
+    AffUsersChanges2 = [{Bob, owner}, {Kate, owner}],
+    escalus:send(Alice, stanza_aff_set(?ROOM, AffUsersChanges2)),
+    verify_aff_bcast([{Alice, owner}, {Bob, owner}, {Kate, owner}], AffUsersChanges2),
+    escalus:assert(is_iq_result, escalus:wait_for_stanza(Alice)),
+    assert_cache_miss_event(TS, 1, room_bin_jid(?ROOM)),
+    assert_cache_hit_event(TS, 1, room_bin_jid(?ROOM))
+                                                           end).
```
Comment (vkatsuba) [inline big_tests/tests/muc_light_SUITE.erl:812]: Please align code block:
```suggestion
multiple_owner_change(Config) ->
    escalus:story(Config, [{alice, 1}, {bob, 1}, {kate, 1}], fun(Alice, Bob, Kate) ->
        TS = instrument_helper:timestamp(),
        AffUsersChanges1 = [{Bob, none}, {Kate, none}],
        escalus:send(Alice, stanza_aff_set(?ROOM, AffUsersChanges1)),
        verify_aff_bcast([{Alice, owner}], AffUsersChanges1),
        escalus:assert(is_iq_result, escalus:wait_for_stanza(Alice)),
        AffUsersChanges2 = [{Bob, owner}, {Kate, owner}],
        escalus:send(Alice, stanza_aff_set(?ROOM, AffUsersChanges2)),
        verify_aff_bcast([{Alice, owner}, {Bob, owner}, {Kate, owner}], AffUsersChanges2),
        escalus:assert(is_iq_result, escalus:wait_for_stanza(Alice)),
        assert_cache_miss_event(TS, 1, room_bin_jid(?ROOM)),
        assert_cache_hit_event(TS, 1, room_bin_jid(?ROOM))
    end).
```

File: doc/modules/mod_muc_light.md [hunk 2]
```diff
@@ -109,6 +109,14 @@ Specifies maximal number of rooms returned for a single Disco request.
 
 When enabled, rooms the user occupies are included in their roster.
 
+### `modules.mod_muc_light.allow_multiple_admin`
+* **Syntax:** boolean
+* **Default:** `true`
```
Comment (vkatsuba) [inline doc/modules/mod_muc_light.md:114]: Not sure but I suppose the default value should be `false`. Those code changes should not affect any previous work or behavior.

File: test/muc_light_SUITE.erl [hunk 3]
```diff
@@ -198,7 +198,7 @@ prop_aff_change_bad_request() ->
     ?FORALL({AffUsers, Changes}, bad_change_aff(),
             begin
                 {error, {bad_request, _}} =
-                    mod_muc_light_utils:change_aff_users(AffUsers, Changes),
+                    mod_muc_light_utils:change_aff_users(<<"localhost">>, AffUsers, Changes),
```
Comment (vkatsuba) [inline test/muc_light_SUITE.erl:201]:
```suggestion
                    mod_muc_light_utils:change_aff_users(?DOMAIN, AffUsers, Changes),
```
or
```suggestion
                    mod_muc_light_utils:change_aff_users(host_type(), AffUsers, Changes),
```

File: test/muc_light_SUITE.erl [hunk 4]
```diff
@@ -164,7 +164,7 @@ filter_room_packet_handler(Acc, _Params, _Extra) ->
 prop_aff_change_success() ->
     ?FORALL({AffUsers, Changes, Joining, Leaving, WithOwner}, change_aff_params(),
             begin
-                case mod_muc_light_utils:change_aff_users(AffUsers, Changes) of
+                case mod_muc_light_utils:change_aff_users(<<"localhost">>, AffUsers, Changes) of
```
Comment (vkatsuba) [inline test/muc_light_SUITE.erl:167]:
```suggestion
                case mod_muc_light_utils:change_aff_users(host_type(), AffUsers, Changes) of
```

File: test/muc_light_SUITE.erl [hunk 5]
```diff
@@ -178,7 +178,7 @@ prop_aff_change_success() ->
                         % are there no owners or there is exactly one?
                         true = validate_owner(NewAffUsers0, false, WithOwner),
                         % changes list applied to old list should produce the same result
-                        {ok, NewAffUsers1, _, _, _} = mod_muc_light_utils:change_aff_users(AffUsers, AffUsersChanged),
+                        {ok, NewAffUsers1, _, _, _} = mod_muc_light_utils:change_aff_users(<<"localhost">>, AffUsers, AffUsersChanged),
```
Comment (vkatsuba) [inline test/muc_light_SUITE.erl:181]:
```suggestion
                        {ok, NewAffUsers1, _, _, _} = mod_muc_light_utils:change_aff_users(host_type(), AffUsers, AffUsersChanged),
```

**Review by vkatsuba — APPROVED** [review]

**Comment by gustawlippa** [comment]
Hello @jaspreet-android, thank you for working on this PR. Thank you also @vkatsuba for taking a look!
I've run the CI, and some tests appear to be failing with these changes. If that's alright with you, I will continue the development in https://github.com/esl/MongooseIM/pull/4392, as CI will not run for a branch from a fork.

**Comment by jaspreet-android** [comment]
> Hello @jaspreet-android, thank you for working on this PR. Thank you also @vkatsuba for taking a look! I've run the CI, and some tests appear to be failing with these changes. If that's alright with you, I will continue the development in #4392, as CI will not run for a branch from a fork.

Yes @gustawlippa it makes sense to me, please go ahead with https://github.com/esl/MongooseIM/pull/4392 as we can check CI there.

**Pull request closed without merging**